sydent/config/__init__.py

```diff
@@ -42,10 +47,17 @@ def _parse_config(self, cfg: ConfigParser) -> None:
         for section in self.config_sections:
             section.parse_config(cfg)
 
-    def parse_from_config_parser(self, cfg: ConfigParser) -> None:
+    def parse_from_config_parser(self, cfg: ConfigParser) -> bool:
         """
         Parse the configuration from a ConfigParser object
 
         :param cfg: the configuration to be parsed
+        ...
+        :return: Whether or not cfg has been changed and needs saving
         """
         self._parse_config(cfg)
+
+        # TODO: Don't alter config file when starting Sydent unless
+        #       user has asked for this specifially (e.g. on first
+        #       run only, or when specify --generate-config)
+        return self.crypto.save_key
```

Comment:
My thought here was that in general, config-files should be able to be read-only files. Therefore, running Sydent shouldn't write to the config file. 

An exception though could be made on first run (or even better only if --generate-config was given as an argument) since in that case, the user is clearly not trying to protect their config file. And so if users want to generate a new signing key, instead of deleting it from the config and restarting sydet, they can use the existing `generate-key` script.

So, my concern with making `parse_config` return a bool was that this would make it easier for other sections to write to the config file.

This does however come at the expense of the less intuitive and quite ugly `crypto.save_key`, which as you say shouldn't persist. 

Anyway if I'm overthinking this then say and I'll just do the bool thing!

Reply:
Have added an abstract base class and made them return bools!